docs/mp/lra.adoc

```diff
@@ -0,0 +1,460 @@
+  participant.url: http://coordinator.visible.host:80/awesomeapp <3>
+----
+<1> Url of coordinator
+<2> Propagate LRA headers LRA_HTTP_CONTEXT_HEADER and LRA_HTTP_PARENT_CONTEXT_HEADER through non-LRA endpoints
+<3> Url of the LRA enabled service overrides standard base uri,
+so coordinator can call load-balancer instead of the service
+
+For more information continue to {microprofile-lra-spec-url}[{spec-name}].
+
+== Examples
+
+The following example shows how a simple LRA participant starts and joins a transaction after calling the '/start-example' resource.
+When startExample method finishes successfully, close is reported to <<Coordinator, coordinator>>
+and `/complete-example` endpoint is called by coordinator to confirm successful closure of the LRA.
+
+If an exception occurs during startExample method execution, coordinator receives cancel call and `/compensate-example`
+is called by coordinator to compensate for cancelled LRA transaction.
+
+[source,java]
+.Example of simple LRA participant
+----
+@PUT
+@LRA(LRA.Type.REQUIRES_NEW) <1>
+@Path("start-example")
+public Response startExample(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId, <2>
+                             String data) {
+    if (data.contains("BOOM")) {
+        throw new RuntimeException("BOOM 💥"); <3>
+    }
+
+    LOGGER.info("Data " + data + " processed 🏭");
+    return Response.ok().build(); <4>
+}
+
+@PUT
+@Complete <5>
+@Path("complete-example")
+public Response completeExample(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+    LOGGER.log(Level.INFO, "LRA id: {0} completed 🎉", lraId);
+    return LRAResponse.completed();
+}
+
+@PUT
+@Compensate <6>
+@Path("compensate-example")
+public Response compensateExample(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+    LOGGER.log(Level.SEVERE, "LRA id: {0} compensated 🦺", lraId);
+    return LRAResponse.compensated();
+}
+----
+<1> This JAX-RS PUT method will start new LRA transactions and join it before method body gets executed
+<2> LRA id assigned by coordinator to this LRA transaction
+<3> When method execution finishes exceptionally, cancel signal for this particular LRA is sent to coordinator
+<4> When method execution finishes successfully, complete signal for this particular LRA is sent to coordinator
+<5> Method which will be called by coordinator when LRA is completed
+<6> Method which will be called by coordinator when LRA is canceled
+
+== Additional Information
+
+=== Coordinator
+Coordinator is a service that tracks all LRA transactions and calls the compensate REST endpoints of
+the participants when the LRA transaction gets cancelled or completes (in case it gets closed).
+In addition, participant also keeps track of timeouts, retries participant calls, and assigns LRA ids.
+
+.Helidon LRA supports following coordinators
+* Helidon LRA coordinator
+* https://narayana.io/lra[Narayana coordinator].
+
+=== Helidon LRA coordinator
+
+CAUTION: Experimental tool, usage in production is not advised.
```

Comment:
Why is this experimental notice in Additional Information? Does this need to be added to the top and repeated? Seems to be relevant earlier in the examples and doc.

Reply:
This is specific only to the coordinator, not the client